src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -150,7 +150,12 @@ public function generateUri($request, $path)
         // fortunately, they all are, so we have to remove entire query string
         $position = strpos($url, '?');
         if (false !== $position) {
```

Comment:
Why is that necessary? What about a path like `/foo/bar#fragment`?

Reply:
Fragment gets stripped only if there is a query string, otherwise the urlGenerator result is returned directly

Reply:
Thanks for the clarification, but maybe we should add a test for that too to prevent regressions?

Reply:
In most cases there is _always_ a query string, because it uses the attributes from the request to build it.

Reply:
@xabbuh Test added